### PR TITLE
Enh binary lines

### PR DESCRIPTION
**Pull request by tacaswell**
so

### CODE REVIEW

File: xray_vision/mpl_plotting.py [hunk 1]
```diff
@@ -120,9 +123,130 @@ def show_label_array(ax, label_array, cmap=None, **kwargs):
...
+
+    Parameters
+    ----------
+    ax : Axes
+        The axes to draw stuff to
+
+    chrom_data : OrderedDict
+        The input data as a dict, keyed on the data label. Each value is
+         with a list of pairs
+        of where the data is 'on'.  Data is plotted top-down
```
Comment (ericdill) [inline xray_vision/mpl_plotting.py]: This doesn't make sense

Comment (ericdill) [inline xray_vision/mpl_plotting.py]: This *docstring

File: xray_vision/mpl_plotting.py [hunk 2]
```diff
@@ -120,9 +123,130 @@ def show_label_array(ax, label_array, cmap=None, **kwargs):
...
                    **kwargs)
 
     return im
+
+
+def binary_state_lines(ax, chrom_data, xmin=0, xmax=120,
+                       delta_y=3,
+                       off_color="#1C2F4D",
+                       on_color="#FA9B00",
+                       lc_kwargs=None):
```
Comment (ericdill) [inline xray_vision/mpl_plotting.py]: Any reason for not defaulting to `None` here?

Comment (danielballan) [inline xray_vision/mpl_plotting.py]: The downside of `None` as a default is that the user can't tell what the defaults actually are without looking at the source code.

File: xray_vision/mpl_plotting.py [hunk 3]
```diff
@@ -120,9 +123,130 @@ def show_label_array(ax, label_array, cmap=None, **kwargs):
...
+    # make the color map and norm
+    cmap = ListedColormap([off_color, on_color])
+    norm = BoundaryNorm([0, 0.5, 1], cmap.N)
+    # dictionary to hold the returned artists
+    ret = dict()
+    # loop over the input data draw each collection
+    for label, data in chrom_data.items():
+        # increment the y offset
+        y_val += delta_y
+        # turn the high windows on to alternating
```
Comment (ericdill) [inline xray_vision/mpl_plotting.py]: what are "high windows"?

File: xray_vision/mpl_plotting.py (hunk 2 above)
Comment (ericdill) [inline xray_vision/mpl_plotting.py]: The docstring of ``lc_kwargs`` tells you what the default for that parameter is and you don't need to look at the source to do that

File: xray_vision/mpl_plotting.py [hunk 4]
```diff
@@ -120,9 +123,130 @@ def show_label_array(ax, label_array, cmap=None, **kwargs):
...
+    x, y, state : array
+       The x edges, the y values and the state of each region
+
+    cmap : matplotlib.colors.Colormap
+       The color map to use
+
+    norm : matplotlib.ticker.Norm
+       The norm to use with the color map
+
+    lw : float, optional
```
Comment (ericdill) [inline xray_vision/mpl_plotting.py]: I would imagine this is supposed to be `lc_kwargs`?

File: xray_vision/mpl_plotting.py (hunk 3 above)
Comment (tacaswell) [inline xray_vision/mpl_plotting.py]: The input data gives you 'windows' where the value is high

Comment (ericdill) [inline xray_vision/mpl_plotting.py]: So this: ?
```
        "high window"    "low window"
      _________________                ________________
_____|                 |______________|                |______________
```

Comment (tacaswell) [inline xray_vision/mpl_plotting.py]: yes

**Comment by ericdill** [comment]
How do you see `binary_state_lines` being used?  As a visual indication of the on/off state of various PVs that during a scan?

**Comment by tacaswell** [comment]
That as well as trying to visualize how asynchronous scans line up for
things with finite exposure.

I also suspect a version of this (probably generalized to more than two
states) is going to be very helpful in debugging the state machine gut in
the run engine refactor.

**Comment by ericdill** [comment]
Would it make sense to provide a syntax that allows you to simply pass in data and low/high limits?  It seems non-trivial to generate the list of tuples that this function expects. Non-trivial because I cannot come up with a way off the top of my head to easily create such a list

**Comment by danielballan** [comment]
zip

**Comment by ericdill** [comment]
I'd like to get this in pretty soon. Could you add a key to show which color is on and which is off?

**Comment by ericdill** [comment]
I rebased and updated the example. It is on ericdill/enh_binary_lines

**Comment by ericdill** [comment]
Suggested improvements:
- add a key with which color corresponds to 0/1 or low/high or on/off. I have no preference in terms of labeling
- have the 'y' value in the lower left corner of the window show the corresponding value. It is currently blank `(x=15.8949 y=)`
- add a time axis at the bottom

I am also totally fine with treating this (awesome) PR as an MVP, merging it as it (after a rebase of course) and turning these suggested improvements into an issue. Thoughts?

**Comment by ericdill** [comment]
Thanks for the rebase. :+1:

**Pull request merged**